llama-index-core/llama_index/core/workflow/workflow.py

```diff
@@ -29,12 +38,28 @@ def __init__(
         self._accepted_events: List[Tuple[str, str]] = []
         self._retval: Any = None
 
+    @classmethod
+    def add_step(cls, func: Callable) -> None:
+        """Adds a free function as step for this workflow instance.
+
+        It raises an exception if a step with the same name was already added to the workflow.
+        """
+        if func.__name__ in {**get_steps_from_class(cls), **cls._step_functions}:
+            msg = f"A step {func.__name__} is already part of this workflow, please choose another name."
+            raise WorkflowValidationError(msg)
+
+        cls._step_functions[func.__name__] = func
+
+    def _get_steps(self) -> Dict[str, Callable]:
```

Comment:
This is just a convenient method to collect all the steps regardless they are methods or free functions